setup.py

```diff
@@ -117,12 +123,21 @@ def __init__(self):
         super(Protobuf, self).__init__()
         # TODO: allow user specify protobuf include_dirs libraries with flags
         use_conda = os.getenv('CONDA_PREFIX') and platform.system() == 'Windows'
+		# Find the path by finding protoc and suppose it's in ${PROTOBUF_PATH}/bin
+        proto_path = which('protoc')
+        if proto_path:
+            proto_path = os.path.realpath(proto_path + "/../")
+
 
         libs = []
+        link_args = []
         if os.getenv('PROTOBUF_LIBDIR'):
             libs.append(os.path.join(os.getenv('PROTOBUF_LIBDIR'), "libprotobuf"))
         elif use_conda:
             libs.append(os.path.join(os.getenv('CONDA_PREFIX'), "Library", "lib", "libprotobuf"))
+        elif proto_path:
+            libs.append(os.path.join(proto_path, "lib", "libprotobuf"))
```

Comment:
The trick is to find `protoc`. For a system with multiple protobuf versions installed, user decides which one to use by setting the `PATH` to pickup the right protoc, and library would be under `../lib`. This is picked up from Caffe2 cmake file and the goal is to be consistent with Caffe2's behavior so that they end up picking the same protobuf.

Reply:
I think https://github.com/onnx/onnx/issues/504 explains a lot on the messed up situation. I can fix that and it will avoid quite a bit customized logic.